src/psyclone/psyir/backend/fortran.py

```diff
@@ -0,0 +1,443 @@
+        '''This method is called when an Array instance is found in the PSyIR
+        tree.
+
+        :param node: An Array PSyIR node.
+        :type node: :py:class:`psyclone.psyGen.Array`
+
+        :returns: The Fortran code as a string.
+        :rtype: str
+
+        '''
+        args = []
+        for child in node.children:
+            args.append(str(self.visit(child)))
+        result = "{0}({1})".format(node.name, ",".join(args))
+        return result
+
+    def literal(self, node):
+        '''This method is called when a Literal instance is found in the PSyIR
+        tree.
+
+        :param node: A Literal PSyIR node.
+        :type node: :py:class:`psyclone.psyGen.Literal`
+
+        :returns: The Fortran code as a string.
+        :rtype: str
+
+        '''
+        result = node._value
+        return result
+
+    def ifblock(self, node):
+        '''This method is called when an IfBlock instance is found in the
+        PSyIR tree.
+
+        :param node: An IfBlock PSyIR node.
+        :type node: :py:class:`psyclone.psyGen.IfBlock`
+
+        :returns: The Fortran code as a string.
+        :rtype: str
+
+        '''
+        condition = self.visit(node.children[0])
+
+        self._depth += 1
+        if_body = ""
+        for child in node.if_body:
+            if_body += self.visit(child)
+        else_body = ""
+        for child in node.else_body:
+            else_body += self.visit(child)
+        self._depth -= 1
+
+        if else_body:
+            result = (
+                "{0}if ({1}) then\n"
+                "{2}"
+                "{0}else\n"
+                "{3}"
+                "{0}end if\n"
+                "".format(self._nindent, condition, if_body, else_body))
+        else:
+            result = (
+                "{0}if ({1}) then\n"
+                "{2}"
+                "{0}end if\n"
+                "".format(self._nindent, condition, if_body))
+        return result
+
+    def unaryoperation(self, node):
+        '''This method is called when a UnaryOperation instance is found in
+        the PSyIR tree.
+
+        :param node: A UnaryOperation PSyIR node.
+        :type node: :py:class:`psyclone.psyGen.UnaryOperation`
+
+        :returns: The Fortran code as a string.
+        :rtype: str
+
+        '''
+        # Reverse the fortran2psyir mapping to make a psyir2fortran
+        # mapping.
+        from psyclone.psyGen import Fparser2ASTProcessor as f2psyir
+        mapping = {}
+        for operator in f2psyir.unary_operators:
+            mapping_key = f2psyir.unary_operators[operator]
+            mapping_value = operator
+            # Only choose the first mapping value when there is more
+            # than one.
+            if mapping_key not in mapping:
+                mapping[mapping_key] = mapping_value
+        oper = mapping[node._operator]
+        content = self.visit(node.children[0])
+        result = "{0}{1}".format(oper, content)
+        return result
```

Comment:
Same than BinaryOperators, there are some operators with functional syntax (.e.g.: sin(x), cos(x))

Reply:
I've done the same thing here as done with binary operators.